corehq/apps/case_search/temp/fix_child_cases.py

```diff
@@ -0,0 +1,85 @@
+def update_script():
+    from corehq.form_processor.models import CommCareCase
+    from corehq.apps.hqcase.case_helper import CaseHelper
+    from corehq.apps.locations.models import SQLLocation
+    import os
+
+    DOMAIN = 'develop'
+    CHILD_CASE_TYPE = 'membre'
+
+    case_ids = CommCareCase.objects.get_case_ids_in_domain(domain=DOMAIN, type=CHILD_CASE_TYPE)
+
+    total_cases = len(case_ids)
+    cases_checked = -1
+
+    error_log_file_path = os.path.expanduser("~/chris_script_error.log")
+    success_log_file_path = os.path.expanduser("~/chris_script_success.log")
+    with open(success_log_file_path, "a") as success_logfile:
+        with open(error_log_file_path, "a") as logfile:
+            for child_case in CommCareCase.objects.iter_cases(case_ids, domain=DOMAIN):
+                # --- Statistics
+                cases_checked += 1
+                progress = round(cases_checked / total_cases, 3)
+                print(f"{progress}% complete")
+                # --------------
+
+                child_case_helper = CaseHelper(domain=DOMAIN, case=child_case)
+                
+                """
+                    Step 2 - Make sure that the case properties of both the child and parent cases correspond to te
+                    village's hierarchy. The owner_id of both cases is the village id
+                """
+                try:
+                    parent_case = child_case.parent
+                    village_id = parent_case.owner_id
+                except Exception:
+                    logfile.write(f"Skipped {child_case.case_id}. Reason: No parent case\n")
```

Comment:
If you use CaseES, you could query only for the cases that have parents, thereby not having to deal with other cases that could slow down the process.